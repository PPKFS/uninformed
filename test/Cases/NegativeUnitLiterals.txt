An elevation is a kind of value. -100 cubits specifies an elevation.
Every room has an elevation.

The water level is an elevation that varies. The water level is -10 cubits.

The first every turn rule:
	say "Water level: [water level].";
	increase the water level by 2 cubits;
	if water level > elevation of location, say "The water has risen to the
		level of [water level]-- above your head!".

When play begins:
	say "Well, assuming this is a logical universe, you should have just
		enough time to finish this ark."

Beach is a room. The elevation of Beach is 2 cubits. A pile of lumber
and a hammer are in Beach.

Every turn when water level is at least elevation of location, end the
story saying "Somehow are you not surprised".

Test me with "z/z/z/z/z/z".

