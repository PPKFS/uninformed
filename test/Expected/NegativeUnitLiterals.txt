133 words
¶an¶ ¶elevation¶ ¶is¶ ¶a¶ ¶kind¶ ¶of¶ ¶value¶ ¶.¶ ¶-100¶ ¶cubits¶ ¶specifies¶ ¶every¶ ¶room¶ ¶has¶ ¶|__¶ ¶the¶ ¶water¶ ¶level¶ ¶that¶ ¶varies¶ ¶-10¶ ¶first¶ ¶turn¶ ¶rule¶ ¶:¶ ¶say¶ ¶"Water level: [water level]."¶ ¶;¶ ¶increase¶ ¶by¶ ¶2¶ ¶if¶ ¶>¶ ¶location¶ ¶,¶ ¶"The water has risen to the
level of [water level]-- above your head!"¶ ¶when¶ ¶play¶ ¶begins¶ ¶"Well, assuming this is a logical universe, you should have just
enough time to finish this ark."¶ ¶beach¶ ¶pile¶ ¶lumber¶ ¶and¶ ¶hammer¶ ¶are¶ ¶in¶ ¶at¶ ¶least¶ ¶end¶ ¶story¶ ¶saying¶ ¶"Somehow are you not surprised"¶ ¶test¶ ¶me¶ ¶with¶ ¶"z/z/z/z/z/z"¶ 
57 distinct words
¶An¶ ¶an¶ 0a
¶elevation¶ ¶elevation¶ 20
¶is¶ ¶is¶ 20
¶a¶ ¶a¶ 20
¶kind¶ ¶kind¶ 20
¶of¶ ¶of¶ 20
¶value¶ ¶value¶ 20
¶.¶ ¶.¶ 20
¶-100¶ ¶-100¶ 20
¶cubits¶ ¶cubits¶ 20
¶specifies¶ ¶specifies¶ 20
¶an¶ ¶an¶ 20
¶elevation¶ ¶elevation¶ 20
¶.¶ ¶.¶ 20
¶Every¶ ¶every¶ 0a
¶room¶ ¶room¶ 20
¶has¶ ¶has¶ 20
¶an¶ ¶an¶ 20
¶elevation¶ ¶elevation¶ 20
¶.¶ ¶.¶ 20
¶|__¶ ¶|__¶ 0a
¶The¶ ¶the¶ 20
¶water¶ ¶water¶ 20
¶level¶ ¶level¶ 20
¶is¶ ¶is¶ 20
¶an¶ ¶an¶ 20
¶elevation¶ ¶elevation¶ 20
¶that¶ ¶that¶ 20
¶varies¶ ¶varies¶ 20
¶.¶ ¶.¶ 20
¶The¶ ¶the¶ 20
¶water¶ ¶water¶ 20
¶level¶ ¶level¶ 20
¶is¶ ¶is¶ 20
¶-10¶ ¶-10¶ 20
¶cubits¶ ¶cubits¶ 20
¶.¶ ¶.¶ 20
¶|__¶ ¶|__¶ 0a
¶The¶ ¶the¶ 20
¶first¶ ¶first¶ 20
¶every¶ ¶every¶ 20
¶turn¶ ¶turn¶ 20
¶rule¶ ¶rule¶ 20
¶:¶ ¶:¶ 20
¶say¶ ¶say¶ 41
¶"Water level: [water level]."¶ ¶"Water level: [water level]."¶ 20
¶;¶ ¶;¶ 20
¶increase¶ ¶increase¶ 41
¶the¶ ¶the¶ 20
¶water¶ ¶water¶ 20
¶level¶ ¶level¶ 20
¶by¶ ¶by¶ 20
¶2¶ ¶2¶ 20
¶cubits¶ ¶cubits¶ 20
¶;¶ ¶;¶ 20
¶if¶ ¶if¶ 41
¶water¶ ¶water¶ 20
¶level¶ ¶level¶ 20
¶>¶ ¶>¶ 20
¶elevation¶ ¶elevation¶ 20
¶of¶ ¶of¶ 20
¶location¶ ¶location¶ 20
¶,¶ ¶,¶ 20
¶say¶ ¶say¶ 20
¶"The water has risen to the
level of [water level]-- above your head!"¶ ¶"The water has risen to the
level of [water level]-- above your head!"¶ 20
¶.¶ ¶.¶ 20
¶|__¶ ¶|__¶ 0a
¶When¶ ¶when¶ 20
¶play¶ ¶play¶ 20
¶begins¶ ¶begins¶ 20
¶:¶ ¶:¶ 20
¶say¶ ¶say¶ 41
¶"Well, assuming this is a logical universe, you should have just
enough time to finish this ark."¶ ¶"Well, assuming this is a logical universe, you should have just
enough time to finish this ark."¶ 20
¶|__¶ ¶|__¶ 0a
¶Beach¶ ¶beach¶ 20
¶is¶ ¶is¶ 20
¶a¶ ¶a¶ 20
¶room¶ ¶room¶ 20
¶.¶ ¶.¶ 20
¶The¶ ¶the¶ 20
¶elevation¶ ¶elevation¶ 20
¶of¶ ¶of¶ 20
¶Beach¶ ¶beach¶ 20
¶is¶ ¶is¶ 20
¶2¶ ¶2¶ 20
¶cubits¶ ¶cubits¶ 20
¶.¶ ¶.¶ 20
¶A¶ ¶a¶ 20
¶pile¶ ¶pile¶ 20
¶of¶ ¶of¶ 20
¶lumber¶ ¶lumber¶ 20
¶and¶ ¶and¶ 0a
¶a¶ ¶a¶ 20
¶hammer¶ ¶hammer¶ 20
¶are¶ ¶are¶ 20
¶in¶ ¶in¶ 20
¶Beach¶ ¶beach¶ 20
¶.¶ ¶.¶ 20
¶|__¶ ¶|__¶ 0a
¶Every¶ ¶every¶ 20
¶turn¶ ¶turn¶ 20
¶when¶ ¶when¶ 20
¶water¶ ¶water¶ 20
¶level¶ ¶level¶ 20
¶is¶ ¶is¶ 20
¶at¶ ¶at¶ 20
¶least¶ ¶least¶ 20
¶elevation¶ ¶elevation¶ 20
¶of¶ ¶of¶ 20
¶location¶ ¶location¶ 20
¶,¶ ¶,¶ 20
¶end¶ ¶end¶ 20
¶the¶ ¶the¶ 20
¶story¶ ¶story¶ 0a
¶saying¶ ¶saying¶ 20
¶"Somehow are you not surprised"¶ ¶"Somehow are you not surprised"¶ 20
¶.¶ ¶.¶ 20
¶|__¶ ¶|__¶ 0a
¶Test¶ ¶test¶ 20
¶me¶ ¶me¶ 20
¶with¶ ¶with¶ 20
¶"z/z/z/z/z/z"¶ ¶"z/z/z/z/z/z"¶ 20
¶.¶ ¶.¶ 20
¶|__¶ ¶|__¶ 0a
¶|__¶ ¶|__¶ 0a
¶|__¶ ¶|__¶ 0a
¶|__¶ ¶|__¶ 0a
¶|__¶ ¶|__¶ 0a
